src/coprocessor/codec/convert.rs

```diff
@@ -463,16 +472,25 @@ mod tests {
 
     #[test]
     fn test_invalid_get_valid_int_prefix() {
+        let mut ctx = EvalContext::new(Arc::new(EvalConfig::default_for_test()));
         let cases = vec!["1e21", "1e9223372036854775807"];
 
+        // Firstly, make sure no error returns, instead a valid float string is returned
         for i in cases {
-            let o = super::float_str_to_int_string(i);
-            assert!(o.is_err());
+            let o = super::get_valid_int_prefix(&mut ctx, i);
+            assert_eq!(o.unwrap(), i);
+        }
+
+        // Secondly, make sure warnings are attached when the float string cannot be casted to a valid int string
+        assert_eq!(ctx.take_warnings().warnings.len(), 2);
```

Comment:
When warnings are taken, you cannot take it again (which will break Line 486). You can consider storing warnings you took in another variable.